### PR TITLE
Use index instead of memmove.

### PR DESCRIPTION
**Pull request by ZehMatt**
I refactored while loop to a proper for loop and removed all the memmove code to use an index instead of shifting everything to 0.

### CODE REVIEW

File: src/openrct2/paint/tile_element/Surface.cpp [hunk 1]
```diff
@@ -641,7 +641,7 @@ static void viewport_surface_draw_tile_side_bottom(paint_session * session, enum
 
     regs.ah = regs.cl;
 
-    while (true)
+    for(uint32 tunelIndex = 0; tunelIndex < TUNNEL_MAX_COUNT; tunelIndex++)
```
Comment (Gymnasiast) [inline src/openrct2/paint/tile_element/Surface.cpp]: tunnelIndex

Comment (ZehMatt) [reply src/openrct2/paint/tile_element/Surface.cpp]: Oh boy

File: src/openrct2/paint/tile_element/Surface.cpp [hunk 2]
```diff
@@ -666,13 +666,12 @@ static void viewport_surface_draw_tile_side_bottom(paint_session * session, enum
         if (curHeight != tunnelArray[0].height)
```
Comment (marijnvdwerf) [inline src/openrct2/paint/tile_element/Surface.cpp]: Missing `tunnelIndex`?

Comment (ZehMatt) [reply src/openrct2/paint/tile_element/Surface.cpp]: Oh yeah

Comment (ZehMatt) [reply src/openrct2/paint/tile_element/Surface.cpp]: It still does it even with tunnelIndex there, there is something I must be missing.

Comment (ZehMatt) [reply src/openrct2/paint/tile_element/Surface.cpp]: Got it, stupid me.

**Pull request merged**